### DISCUSSION TITLE
Improve Helm Chart's values

### DISCUSSION THREAD

**Discussion by Ramblurr** [nextcloud/all-in-one#2589] [Ideas] [answered]
The helm chart is somewhat limited when it comes to the configuration options:

* The PVC names are static, which makes deploying more than one instance into the same namespace not feasible
* The various deployments all use the same `STORAGE_CLASS`. But in a real production instance one would want to use different storage classes for different types of data.
* The helm chart assumes  the user wants to expose the service via a `LoadBalancer`. But in production k8s clusters we have ingress' already configure that provide (amongst other things) SSL termination.

## Potential improvements

* Configuration values for:
  * the name of the PVCs
  * different `*_STORAGE_CLASS` values (i.e., nextcloud, nextcloud data, etc)
* The ability to configure an ingress with an `ingressClassName`, annotations, etc. 

IMO these improvements are necessary before the chart could be considered production ready.

They are also just the first things I noticed when trying to deploy the chart, there are also probably a bunch of other possible improvements.

Having looked at how the chart is produced, I'm not sure if these changes can be made easily (though I'm not familiar with the kompose tool).

Is there interest in improving this? I would be happy to contribute if there is interest from the maintainers in improving the helm chart.

**Comment by szaimen** [comment, marked as answer]
Hi, so in general we are of course interested in improving the helm chart. Otherwise it would not exist in the first place 😉

However we want to keep it as simple as possible in order to make the first-time-experience for newcomers as straight-forward as possible. So only really necessary changes will be accepted. If you need more customizability, refer to https://github.com/nextcloud/helm.

So before contributing, please ask yourself if the change is really necessary or if there are some workarounds that you could use.

Now that this is out of the way, I do want to mention that I will not be able to work on this in the coming months. So contributions are definitely welcome! :) I've added you to the contribution list in the meanwhile so that collaboration is easier for you and us. 

Thanks a lot in advance!

PS: Changes on the helm chart will need to be automated by adding them to this script: https://github.com/nextcloud/all-in-one/blob/main/nextcloud-aio-helm-chart/update-helm.sh. If you have questions regarding this, feel free to ask!

**Reply by szaimen** [reply]
@Ramblurr would you be up for implementing the requested changes if you need them?

**Comment by szaimen** [comment]
Some questions regarding the feature requests from my side:
> * The PVC names are static, which makes deploying more than one instance into the same namespace not feasible

So why not use a different namespace then?

> * The various deployments all use the same `STORAGE_CLASS`. But in a real production instance one would want to use different storage classes for different types of data.

I get the need for this but maybe things could be separated into 2 or three customizable storage classes for the different volumes? E.g. some that really need block storage, some that are fine with network storage, etc.?

> * The helm chart assumes  the user wants to expose the service via a `LoadBalancer`. But in production k8s clusters we have ingress' already configure that provide (amongst other things) SSL termination.

Is there really no way to e.g. configure the incress to point to a specific port?

**Comment by szaimen** [comment]
@Ramblurr here is a good example why we automated building the helm chart: https://github.com/nextcloud/all-in-one/pull/2748

**Reply by szaimen** [reply]
And here is another one: https://github.com/nextcloud/all-in-one/pull/2864
